pkg/R/RDD.R

```diff
@@ -1294,6 +1294,79 @@ setMethod("sortBy",
             values(sortByKey(keyBy(rdd, func), ascending, numPartitions))
           })
 
+# Helper function to get first N elements from an RDD in the specified order.
+# Param:
+#   rdd An RDD.
+#   num Number of elements to return.
+#   ascending A flag to indicate whether the sorting is ascending or descending.
+# Return:
+#   A list of the first N elements from the RDD in the specified order.
+#
+takeOrderedElem <- function(rdd, num, ascending = TRUE) {          
+  if (num <= 0L) {
+    return(list())
+  }
+  
+  partitionFunc <- function(part) {
+    if (num < length(part)) {
+      # R limitation: order works only on primitive types!
+      ord <- order(unlist(part, recursive = FALSE), decreasing = !ascending)
+      part[ord[1:num]]
+    } else {
+      part
+    }
+  }
+  
+  newRdd <- mapPartitions(rdd, partitionFunc)
+  take(sortBy(newRdd, function(x) { x }, ascending = ascending), num)
```

Comment:
Couldn't we just do a collect on `newRdd` and then take the top `num` ? It looks like the scala implementation is doing something similar [1] and it will save the cost of shuffling in `sortBy`

[1] https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/rdd/RDD.scala#L1237